x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -66,7 +66,7 @@ public void clusterChanged(ClusterChangedEvent event) {
         // The atomic flag prevents multiple simultaneous attempts to create the
         // index if there is a flurry of cluster state updates in quick succession
         if (event.localNodeMaster() && isIndexCreationInProgress.compareAndSet(false, true)) {
-            AnnotationIndex.createAnnotationsIndexIfNecessary(settings, client, event.state(), ActionListener.wrap(
+            AnnotationIndex.createAnnotationsIndexIfNecessary(client, event.state(), ActionListener.wrap(
```

Comment:
This still leaves open the following bug (unsure how to fix it really). 

- Node running the job is updated 
- Job gets reallocated to a new node
- Job writes annotation for a model snapshot
- The master node is still Old, consequently never crated the annotations index
- Annotation persistence auto-created the index and now the UI fails to read from the index.

It seems to me that we need to verify if the index exists already and create it if necessary when we write annotations.

Reply:
I think that scenario was a problem in 6.7 and 6.8 as there were 6.x indices that didn't have the annotations index - it probably explains a few bug reports we had when the annotations index was not available as expected.  But now any rolling upgrades must be starting from 6.8 at the earliest, so the old version will have the annotations index already.

Reply:
OK, my concern is that this will occur again whenever we update the mapping of the annotation index. 

We can sort of 'boot' that for now I suppose. But we should definitely make sure that whomever updates the mapping of the annotations index knows that they need make this sort of change.

Reply:
> OK, my concern is that this will occur again whenever we update the mapping of the annotation index.

My PR does not update mappings of annotations index. 
The new annotations I'm adding fit the existing mappings so there was no point in changing them.